src/Fmt_ast.ml

```diff
@@ -1915,12 +1931,10 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
       in
       match compact with
       | None ->
-          let leading_cmt = Cmts.fmt_before c e0.pexp_loc in
```

Comment:
This will cause regressions on OCaml 4.07 but improve things on 4.08.
Not sure what to do here ..

Reply:
I did not find an example. Could you give one?

Reply:
```
let () = ( (* before *) match (* after *) x with _ -> x)
```

with 4.07
```
let () = match (* before *)
               (* after *) x with _ -> x
```

with 4.08
```
let () = (* before *) match (* after *) x with _ -> x
```

Reply:
Thanks. I got surprised by the `dune-workspace` file.

This bug is also present (but comments are placed before) on `master` with 4.07, so at least it's not a regression:

```diff
-let () = ( (* before *) match (* after *) x with _ -> x)
+let () = (* before *)
+         (* after *) match x with _ -> x
```

Reply:
Well, comments were place before on purpose to mitigate the issue. I believe it will be perceived as a regression by people stuck with 4.07.
IIRC, the main issue was

```ocaml
(* long comment
    on mutli
    lines
*)
match x with
| A -> 1
| B -> 0
```

that got reformatted into

```ocaml
match
  (* long comment
    on mutli
    lines
  *)
  x
with
| A -> 1
| B -> 0
```

Reply:
Reverting this change doesn't seems to add regressions on 4.08. Maybe we could open an issue to solve this later?